python/rikai/tf/data.py

```diff
@@ -0,0 +1,54 @@
+#  Copyright 2022 Rikai Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Tensorflow Dataset"""
+
+from pathlib import Path
+from typing import List, Optional, Tuple, Union
+
+import tensorflow as tf
+
+from rikai.parquet import Dataset
+from rikai.parquet.dataset import convert_tensor
+
+__all__ = ["from_rikai"]
+
+
+def from_rikai(
+    data_ref: Union[str, Path],
+    output_signature: Tuple,
+    columns: Optional[List[str]] = None,
+):
+    """Build a Tensorflow (tf) Dataset from rikai dataset.
+
+    Parameters
+    ----------
+    query : str
+        A dataset URI
+    output_signature : tuple
```

Comment:
Is the plan that we will try to infer the schema in the future?

Reply:
Really wanted to do so. Tried a few days earlier last year, don't see a good elegant solution without users either especially specifying the `schema / spec / shape` either in producer side (in feature engineering) or consumer side (data loader).

The main mismatch is that I could not find a way to define parameterized user defined types in Spark, so that it is hard to persist shapes, especially for the `Numpy array` type in the dataset.  

So instead, if we wanted to persist the schema in the producer side, we can probably do something like

```
df.write.format("rikai).option("rikai.data.spec", SomeSpec.to_json()).save("s3://foo/bar")
```